scheduler/src/cook/rest/api.clj

```diff
@@ -730,7 +738,14 @@
                                                                 (str/lower-case operator))
                                   :constraint/pattern pattern}]))
                             constraints)
-        container (if (nil? container) [] (build-container user db-id container))
+        pool-name (or (:pool/name pool) (config/default-pool))
+        default-container-for-pool (get-in config/config [:settings :pools :default-container-for-pool])
+        container (if (nil? container)
+                    (let [guessed-container (match-default-container-for-pool default-container-for-pool pool-name)]
+                      (if (and pool-name guessed-container)
```

Comment:
Why are we checking `pool-name` here?

Reply:
If we're not running with a pools, then there's no pool to match guessed containers. This makes the logic explicit, instead of implicit, that without pools, we are not using [:config :pools :default-containers] and keeps nil's out of that code.

Reply:
Ah, got it, makes sense. In this case (`pool-name` is `nil`), should we just avoid making the call to `match-default-container-for-pool`?

Reply:
Ok.